datadog/dogshell/wrap.py

```diff
@@ -219,25 +221,42 @@ def build_event_body(cmd, returncode, stdout, stderr, notifications):
         )
 
 
+def generate_warning_codes(option, opt, options_warning):
+    try:
+        # options_warning should be a comma separated string
+        warning_codes = [options_warning.split(",")]
+        return warning_codes
+    except ValueError:
+        raise optparse.OptionValueError("option %s: invalid warning codes value(s): %r" % (opt, options_warning))
+
+
+class WarningOption(optparse.Option):
+    TYPES = optparse.Option.TYPES + ("warning_codes",)
+    TYPE_CHECKER = copy(optparse.Option.TYPE_CHECKER)
+    TYPE_CHECKER["warning_codes"] = generate_warning_codes
+
+
 def parse_options(raw_args=None):
     '''
     Parse the raw command line options into an options object and the remaining command string
     '''
     parser = optparse.OptionParser(usage="%prog -n [event_name] -k [api_key] --submit_mode \
-[ all | errors ] [options] \"command\". \n\nNote that you need to enclose your command in \
+[ all | errors | warnings] [options] \"command\". \n\nNote that you need to enclose your command in \
 quotes to prevent python executing as soon as there is a space in your command. \n \nNOTICE: In \
 normal mode, the whole stderr is printed before stdout, in flush_live mode they will be mixed but \
 there is not guarantee that messages sent by the command on both stderr and stdout are printed in \
-the order they were sent.", version="%prog {0}".format(get_version()))
+the order they were sent.", version="%prog {0}".format(get_version()), option_class=WarningOption)
 
     parser.add_option('-n', '--name', action='store', type='string', help="the name of the event \
 as it should appear on your Datadog stream")
     parser.add_option('-k', '--api_key', action='store', type='string',
                       help="your DataDog API Key")
     parser.add_option('-m', '--submit_mode', action='store', type='choice',
-                      default='errors', choices=['errors', 'all'], help="[ all | errors ] if set \
+                      default='errors', choices=['errors', 'warnings', 'all'], help="[ all | errors | warnings ] if set \
 to error, an event will be sent only of the command exits with a non zero exit status or if it \
-times out.")
+times out. If set to warning, a list of exit codes need to be provided")
+    parser.add_option('--warning_codes', action='store', type='warning_codes', dest='warning_codes', default='',
```

Comment:
That's where you could use your `WarningOption` class, but I'm not sure it's needed.

Using `type=warning_codes` should be enough.

(ok I admit I forgot optparse but I expect it to be close to argparse :))

Reply:
> Using type=warning_codes should be enough.

Yes `WarningOption` should be defined at the parser global config level.